### PR TITLE
Enable Arm unit test when running pull jobs

### PR DESCRIPTION
**Pull request by zingo**
Trigger and run backend arm tests as par of github ci

### CODE REVIEW

**Comment by zingo** [comment]
There are some different way to do this and maybe we could also limit scope of testing to only arm backend test for this job to make it faster, please feel free to comment suggest on your thoughts. :)

File: .github/workflows/pull.yml [hunk 1]
```diff
@@ -243,3 +243,39 @@ jobs:
...
+        # The generic Linux job chooses to use base env, not the one setup by the image
+        CONDA_ENV=$(conda env list --json | jq -r ".envs | .[-1]")
+        conda activate "${CONDA_ENV}"
+
+        BUILD_TOOL=${{ matrix.build-tool }}
+
+        # Setup MacOS dependencies as there is no Docker support on MacOS atm
+        PYTHON_EXECUTABLE=python \
+        EXECUTORCH_BUILD_PYBIND=ON \
+        EXECUTORCH_BUILD_XNNPACK=ON \
```
Comment (digantdesai) [inline .github/workflows/pull.yml:271]: Curious, why do we need XNNPACK?

File: .github/workflows/pull.yml [hunk 2]
```diff
@@ -243,3 +243,39 @@ jobs:
...
+        EXECUTORCH_BUILD_ARM_BAREMETAL=ON \
+        .ci/scripts/setup-linux.sh "${BUILD_TOOL}"
+
+        source .ci/scripts/utils.sh
+        # Install Arm dependencies
+        install_arm
+
+        # Run pytest with coverage
+        pytest -c /dev/null -v -n auto --cov=./ --cov-report=xml backends/arm/test
```
Comment (digantdesai) [inline .github/workflows/pull.yml:280]: This shouldn't run FVP jobs so should be relatively quick right? I am ok with this. CC @guangy10

Comment (digantdesai) [reply .github/workflows/pull.yml:343]: @zingo  - how long do these tests take on CI? Just curious

**Comment by digantdesai** [comment]
Nothing from my side, we are discussing internally but hopefully should resolve today.

**Comment by zingo** [comment]
Anything we can change/update in the patch to improve the chance of a fruitful outcome? 😊

File: .github/workflows/pull.yml (hunk 2 above)
Comment (digantdesai) [reply .github/workflows/pull.yml:343]: Let's enable this and we will disable if something doesn't work right

**Review by digantdesai — APPROVED** [review]

**Review by digantdesai — APPROVED** [review]

File: .ci/scripts/utils.sh [hunk 3]
```diff
@@ -89,6 +90,17 @@ install_flatc_from_source() {
   popd || return
 }
 
+install_arm() {
```
Comment (digantdesai) [inline .ci/scripts/utils.sh:93]: Nit: Since we are using relative paths, it might be a good ideas to make sure we are indeed in the right dir.

**Comment by digantdesai** [comment]
Trying to merge now.

File: .github/workflows/pull.yml [hunk 4]
```diff
@@ -307,3 +307,38 @@ jobs:
...
+
```
Comment (digantdesai) [inline .github/workflows/pull.yml:344]: Internal linter is complaining about this newline.
@zingo can you please remove this and also rebase, push again? 🙏

**Pull request closed without merging**